### PR TITLE
[Docs] Non-Breaking change of "p4c" to "P4C"

### PR DESCRIPTION
**Pull request by AdarshRawat1**
# Fixes 
-  #4585

# Changes 
Refactor : Current Compiler Acronym i.e. `p4c` to `P4C`  in Documentation

> This PR only focuses on the Documentation text.
> Without modifying any link, command, directory name and "p4c" Repository name .

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -155,20 +155,20 @@ sudo apt update
...
 ```
 
-If you cannot use a repository to install p4c, you can download the `.deb` file
+If you cannot use a repository to install P4C, you can download the `.deb` file
 for your release and install it manually. You need to download a new file each
-time you want to upgrade p4c.
+time you want to upgrade P4C.
 
-1. Go to [p4lang-p4c package page on OpenSUSE Build Service](https://build.opensuse.org/package/show/home:p4lang/p4lang-p4c), click on
+1. Go to [p4lang-P4C package page on OpenSUSE Build Service](https://build.opensuse.org/package/show/home:p4lang/p4lang-p4c), click on
```
Comment (fruffy) [inline README.md:162]:
```suggestion
1. Go to [p4lang-p4c package page on OpenSUSE Build Service](https://build.opensuse.org/package/show/home:p4lang/p4lang-p4c), click on
```

File: README.md [hunk 2]
```diff
@@ -238,7 +238,7 @@ sudo dpkg -i /path/to/package.deb
     ```
     sudo make install
     ```
-    The compiler driver `p4c` and binaries for each of the backends are
+    The compiler driver `P4C` and binaries for each of the backends are
```
Comment (fruffy) [inline README.md:241]:
```suggestion
    The compiler driver `p4c` and binaries for each of the backends are
```

File: README.md [hunk 3]
```diff
@@ -309,13 +309,13 @@ After cloning Protobuf and before you build, check-out version 3.25.3:
 `git checkout v3.25.3`
 
 Please note that while all Protobuf versions newer than 3.0 should work for
-`p4c` itself, you may run into trouble with some extensions and other p4lang
+`P4C` itself, you may run into trouble with some extensions and other p4lang
```
Comment (fruffy) [inline README.md:312]:
```suggestion
P4C itself, you may run into trouble with some extensions and other p4lang
```

File: README.md [hunk 4]
```diff
@@ -309,13 +309,13 @@ After cloning Protobuf and before you build, check-out version 3.25.3:
...
 projects unless you install version 3.25.3.
 
-`p4c` also depends on Google Abseil library. This library is also a pre-requisite for Protobuf of any version newer than 3.21. Therefore the use of Protobuf of suitable version automatically fulfils Abseil dependency. P4C typically installs its own version of Abseil using CMake's `FetchContent` module (Abseil LTS 20240116.1 at the moment).
+`P4C` also depends on Google Abseil library. This library is also a pre-requisite for Protobuf of any version newer than 3.21. Therefore the use of Protobuf of suitable version automatically fulfils Abseil dependency. P4C typically installs its own version of Abseil using CMake's `FetchContent` module (Abseil LTS 20240116.1 at the moment).
```
Comment (fruffy) [inline README.md:315]:
```suggestion
P4C also depends on Google Abseil library. This library is also a pre-requisite for Protobuf of any version newer than 3.21. Therefore the use of Protobuf of suitable version automatically fulfils Abseil dependency. P4C typically installs its own version of Abseil using CMake's `FetchContent` module (Abseil LTS 20240116.1 at the moment).
```

File: docs/README.md [hunk 5]
```diff
@@ -292,7 +292,7 @@ To add a new input test with a sample P4 code file (under `testdata/p4_16_sample
 
 ## Compiler Driver
 
-**p4c** is a compiler driver. The goal is to provide a consistent user interface
+**P4C** is a compiler driver. The goal is to provide a consistent user interface
```
Comment (fruffy) [inline docs/README.md:295]:
```suggestion
**p4c** is a compiler driver. The goal is to provide a consistent user interface
```

File: docs/README.md [hunk 6]
```diff
@@ -335,7 +335,7 @@ makefile variable.
 p4c_PYTHON += p4c.custom.cfg
 ```
 
-There is an global variable `config` in p4c compiler driver that stores the build steps
+There is an global variable `config` in P4C compiler driver that stores the build steps
```
Comment (fruffy) [inline docs/README.md:338]:
```suggestion
There is an global variable `config` in the `p4c` compiler driver that stores the build steps
```

File: docs/README.md [hunk 7]
```diff
@@ -354,5 +354,5 @@ config.target.append("bmv2-newarch-p4org")
 
 After adding the new configuration file, rerun `bootstrap.sh`
 
-For testing purpose, p4c will be installed in the build/ directory when executing `make`.
-User can install `p4c` to other system path by running `make install`
+For testing purpose, P4C will be installed in the build/ directory when executing `make`.
+User can install `P4C` to other system path by running `make install`
```
Comment (fruffy) [inline docs/README.md:358]:
```suggestion
For testing purposes, `p4c` will be installed in the build/ directory when executing `make`.
Users can install `p4c` to other system path by running `make install`
```

**Review by fruffy — CHANGES_REQUESTED** [review]

**Review by fruffy — APPROVED** [review]

**Comment by fruffy** [comment]
Is this ready to be merged?

**Comment by AdarshRawat1** [comment]
According to me it is ready, rest of the sitings are directory name and links.

**Comment by AdarshRawat1** [comment]
Are we planning to change the repository name in near future?

**Comment by fruffy** [comment]
> According to me it is ready, rest of the sitings are directory name and links.

I was asking because you left 


    Work in progress 🚧🚀

in the description. 

>Are we planning to capitalise the repository name in near future?

No, that will stay lower case.

**Comment by AdarshRawat1** [comment]
> I was asking because you left
> 
> ```
> Work in progress 🚧🚀
> ``` 
> in the description.

I had doubts if the repository name will change or not. Since it won't change, this PR is ready to go.

**Pull request merged**